addons/pos_self_order/controllers/qr_menu.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+
+from typing import Dict, Optional
+import werkzeug
+
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.pos_self_order.controllers.utils import (
+    get_pos_config_sudo,
+    get_any_pos_config_sudo,
+    get_table_sudo,
+)
+from odoo.addons.pos_self_order.models.pos_config import PosConfig
+from odoo.addons.pos_self_order.models.pos_restaurant import RestaurantTable
+
+
+class PosQRMenuController(http.Controller):
+    """
+    This is the controller for the POS Self Order App
+    There is one main route that the client will use to access the POS Self Order App: /menu
+    This route will render the LANDING PAGE of the POS Self Order App
+    And it will pass the needed data to the template: the list of products, pos_config_id, table_id, company_name, currency...
+    After that the client will be able to navigate to the /products route w/o aditional requests
+    to the server, using client side routing.
+    """
+
+    @http.route("/menu", auth="public")
+    def pos_self_order_redirect(self):
+        return request.redirect(f"/menu/{get_any_pos_config_sudo().id}")
+
+    @http.route(
+        [
+            "/menu/<pos_name>",
+            "/menu/<pos_name>/products",
+            "/menu/<pos_name>/products/<int:product_id>",
+            "/menu/<pos_name>/cart",
+            "/menu/<pos_name>/orders",
+        ],
+        auth="public",
+        website=True,
+        sitemap=True,
+    )
+    def pos_self_order_start(
+        self, pos_name: str, id: Optional[str] = None, product_id: Optional[int] = None
+    ):
+        """
+        The user gets this route from the QR code that they scan at the table
+        :param pos_name: the name of the pos config: can be the id or the slugified name of the pos config. (e.g. "3" or "bar-3")
+        :param id: the access token of the table; we call this argument "id" because
+            it will be displayed in the url ( as a query param ), and "id" is more user friendly than "table_access_token"
+            the user is allowed to order only if this "id" matches the access token of a table
+        :param product_id: the id of the product that the user wants to see the details of;
+            we never actually use this argument in this function ( it will be read by the client side router ),
+            but we still have it here, because otherwise we get a Warning in the logs
+        :return: the rendered template
+        """
+        return request.render(
+            "pos_self_order.index",
+            {
+                "session_info": {
+                    **request.env["ir.http"].get_frontend_session_info(),
+                    "currencies": request.env["ir.http"].get_currencies(),
+                    "pos_self_order": self._get_self_order_config(
```

Comment:
```suggestion
                    "pos_self_order_data": self._get_self_order_config(
```

Can you rename it to `pos_self_order_data`? It kinda makes more sense. Besides, I think you'll only replace it only in 2 places.